array.py

```diff
@@ -27,7 +27,7 @@ def f_contiguous_strides(itemsize, shape):
     if shape:
         strides = [itemsize]
         for s in shape[:-1]:
-            strides.append(strides[-1]*s)
+            strides.append(strides[-1]*max(1, s))
```

Comment:
Add a comment justifying this? (Below also.) The approach is OK: The stride is irrelevant for a any shape axis <= 1.

Reply:
> such that the result is consistent with `np.empty(shape).strides`

Are you sure of this? If not, we shouldn't claim it. Personally, I'm more comfortable with "the stride doesn't matter, and a zero stride makes a lot of code unhappy (provide example)".